specs/_features/maxeb_increase/capella.py

```diff
@@ -1146,10 +1148,19 @@ def initiate_validator_exit(state: BeaconState, index: ValidatorIndex) -> None:
     # Compute exit queue epoch
     exit_epochs = [v.exit_epoch for v in state.validators if v.exit_epoch != FAR_FUTURE_EPOCH]
     exit_queue_epoch = max(exit_epochs + [compute_activation_exit_epoch(get_current_epoch(state))])
-    exit_queue_churn = len([v for v in state.validators if v.exit_epoch == exit_queue_epoch])
-    if exit_queue_churn >= get_validator_churn_limit(state):
-        exit_queue_epoch += Epoch(1)
-
+    if exit_queue_epoch > max(exit_epochs): 
+        state.exit_queue_churn = Gwei(0)
+    churn_limit = get_validator_churn_limit(state)
+    if state.exit_queue_churn + validator.effective_balance <= churn_limit:  # The validator fits within the churn of the current exit_queue_epoch
+        state.exit_queue_churn += validator.effective_balance  # The full effective balance of the validator contributes to the churn in the exit queue epoch 
+    else:  # The validator does not fit within the churn of the current exit_queue_epoch
+        future_epochs_churn_contribution = validator.effective_balance - (churn_limit - state.exit_queue_churn)
+        exit_queue_epoch += Epoch((future_epochs_churn_contribution + churn_limit - 1) // churn_limit)  # (numerator + denominator - 1) // denominator rounds up
+        # The validator contributes to the churn in the exit queue epoch, based on how much balance is left over at that point 
+        if future_epochs_churn_contribution % churn_limit == 0:
+            state.exit_queue_churn = churn_limit
+        else:
+            state.exit_queue_churn = future_epochs_churn_contribution % churn_limit
```

Comment:
This whole chunk of code may be significantly more readable with a `while` loop.